File: app/models/department.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base

class Department(Base):
    __tablename__ = "departments"

    department_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    department_name = Column(String(100), nullable=False)
    description = Column(String(255), nullable=True)
    status = Column(String(20), default="Active", nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    # Relationships
    employees = relationship("Employee", back_populates="department")

File: web-admin/src/types/department.ts
export interface Department {
  department_id: number;
  department_name: string;
  description?: string;
  status: string;
  created_at?: string;
  updated_at?: string;
}

export interface DepartmentCreate {
  department_name: string;
  description?: string;
  status?: string;
}
